src/containers/Layout/Layout.tsx

```diff
@@ -48,17 +53,17 @@ const Layout: FC<LayoutProps> = ({ children }) => {
   }, [searchActive]);
 
   const searchButtonClickHandler = () => {
-    UIStore.update((s) => {
-      s.searchActive = true;
-      s.preSearchPage = history.location;
+    useUIStore.setState({
+      searchActive: true,
+      preSearchPage: history.location,
```

Comment:
Minor, but we could create a setter in UIStore which leads to `setSearchActive(true, history.location)`